target/linux/ramips/patches-4.4/997-ralink-Introduce-fw_passed_dtb-to-arch-mips-ralink.patch

```diff
@@ -0,0 +1,28 @@
+--- a/arch/mips/ralink/of.c
++++ b/arch/mips/ralink/of.c
+@@ -81,13 +81,23 @@
+ 
+ void __init plat_mem_setup(void)
+ {
++	void *dtb;
++
+ 	set_io_port_base(KSEG1);
+ 
+ 	/*
+ 	 * Load the builtin devicetree. This causes the chosen node to be
+ 	 * parsed resulting in our memory appearing. fw_passed_dtb is used
+	 * by CONFIG_MIPS_APPENDED_RAW_DTB as well.
+ 	 */
+-	__dt_setup_arch(&__image_dtb);
++	if (fw_passed_dtb)
++		dtb = (void *)fw_passed_dtb;
++	else if (__dtb_start != __dtb_end)
++		dtb = (void *)__dtb_start;
++	else
++		dtb = &__image_dtb;
++
++	if (dtb)
++		__dt_setup_arch(dtb);
```

Comment:
I guess that change would break at least one platform because as can be seen at [this](http://lxr.free-electrons.com/source/arch/mips/ralink/of.c?v=4.4#L78) source code line even ralink is intended to default to two different variants of other information from soc_info to initiate even without a DTB which worked for me for rt288x as well after fixing the above linked bug.

For platforms really needing DTB to work (even ralink ones) a check is part of "__dt_register_buses()" [[1](http://lxr.free-electrons.com/source/arch/mips/kernel/prom.c?v=4.4#L64)] that panics.

Based on the above check in __dt_setup_arch() I guess we can leave out the if condition all together as this function should be able to handle a null value for dtb.